CI: query_counts.json baseline stale against NetBox main (tests (main) failing)

## Summary

The `tests (main)` CI matrix leg is currently failing on an unmodified `main` checkout due to stale query-count baselines in `netbox_custom_objects/tests/query_counts.json`. This is unrelated to any specific PR's changes — it reproduces on `main` as-is.

## Details

`tests (main)` runs against NetBox's `main` git ref, fetched fresh at CI time rather than pinned to a commit (see `.github/workflows/lint-tests.yaml`). Something on `netbox-community/netbox`'s `main` branch has apparently shaved 2 queries off several list-view permission checks since these baselines were last recorded, causing `assert_expected_query_count()` (`netbox/utilities/testing/query_counts.py`) to fail with a mismatch on 4 tests:

```
FAIL: test_list_objects_with_permission (netbox_custom_objects.tests.test_views.ComplexCustomObjectViewTestCase.test_list_objects_with_permission)
AssertionError: Query count for netbox_custom_objects/customobject-complex:list_objects_with_permission changed: expected 41, got 39.

FAIL: test_list_objects_with_permission (netbox_custom_objects.tests.test_views.CustomObjectTypeViewTestCase.test_list_objects_with_permission)
AssertionError: Query count for netbox_custom_objects/customobjecttype:list_objects_with_permission changed: expected 34, got 32.

FAIL: test_list_objects_with_permission (netbox_custom_objects.tests.test_views.CustomObjectViewTestCase.test_list_objects_with_permission)
AssertionError: Query count for netbox_custom_objects/customobject-simple:list_objects_with_permission changed: expected 33, got 31.

FAIL: test_list_objects_with_permission (netbox_custom_objects.tests.test_views.ObjectFieldViewTestCase.test_list_objects_with_permission)
AssertionError: Query count for netbox_custom_objects/customobject-objectfields:list_objects_with_permission changed: expected 50, got 48.
```

Confirmed via a `workflow_dispatch` run of `lint-tests.yaml` against unmodified `main` (run [31199832224](https://github.com/netboxlabs/netbox-custom-objects/actions/runs/31199832224)) — same 4 tests fail with the identical deltas. `tests (feature)` and `tests (main, branching)` both pass (branching skips the query-count assertion entirely; `feature`'s NetBox ref apparently hasn't picked up whatever changed on `main` yet).

Originally noticed as an unrelated CI failure on PR #648.

## Fix

Re-run the suite serially against current NetBox `main` with `UPDATE_QUERY_COUNTS=1` to regenerate `netbox_custom_objects/tests/query_counts.json`, then verify the new counts are a genuine (not accidental) reduction — i.e. confirm no query-count *increase* elsewhere was masked in the same run — before committing the refreshed baseline.

```bash
UPDATE_QUERY_COUNTS=1 python netbox/manage.py test netbox_custom_objects.tests.test_views --keepdb
```

--- a/netbox_custom_objects/tests/query_counts.json
+++ b/netbox_custom_objects/tests/query_counts.json
@@ -1,6 +1,6 @@
 {
-  "customobject-complex:list_objects_with_permission": 41,
-  "customobject-objectfields:list_objects_with_permission": 50,
-  "customobject-simple:list_objects_with_permission": 33,
-  "customobjecttype:list_objects_with_permission": 34
+  "customobject-complex:list_objects_with_permission": 39,
+  "customobject-objectfields:list_objects_with_permission": 48,
+  "customobject-simple:list_objects_with_permission": 31,
+  "customobjecttype:list_objects_with_permission": 32
 }

